internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,6 +91,118 @@ func TestDelayConsumer_Run(t *testing.T) {
+	msg1 := getDummyBrokerMessage()
+	msg1.NextDeliveryTime = time.Now().Add(time.Nanosecond * 1000)
+	msg2 := getDummyBrokerMessage()
+	msg2.NextDeliveryTime = time.Now().Add(time.Second * 2)
+	msgs = append(msgs, msg1, msg2)
+	type args struct {
+		messages []messagebroker.ReceivedMessage
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "Test Delay Consumer one message counsume out of 2",
+			wantErr: false,
+			args: args{
+				messages: msgs,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			count := 0
+			consumer.EXPECT().ReceiveMessages(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: tt.args.messages}, nil
+					} else {
+						return &messagebroker.GetMessagesFromTopicResponse{}, nil
+					}
+				}).AnyTimes()
+			go dc.Run(ctx)
+			select {
+			case err := <-errCh:
+				if !tt.wantErr {
+					t.Errorf("Got Error : %v", err)
+				}
+			case <-time.NewTicker(time.Millisecond * 500).C:
+				assert.Equal(t, len(dc.cachedMsgs), 1)
+				cancel()
+				<-time.NewTicker(time.Millisecond * 1).C
+			}
+		})
+	}
+}
+
+func TestDelayConsumer_Run_DeadLetter(t *testing.T) {
+	ctx, subCtx, cancel, store, handler, consumer, cache := setupDC(t)
+	subs := getDummySubscriptionModel()
+	subscriberID := "subscriber-dl-id"
+	errCh := make(chan error)
+	dc, err := NewDelayConsumer(subCtx, subscriberID, "t1-dl", subs, store, handler, cache, errCh)
+	assert.NotNil(t, dc)
+	assert.Nil(t, err)
+	assert.NotNil(t, dc.LogFields())
+	tFut := time.Now().Add(time.Second * 10)
+	msgs := make([]messagebroker.ReceivedMessage, 0)
+	msg := getDummyBrokerMessageDLQ()
+	msg.NextDeliveryTime = tFut
+	msgs = append(msgs, msg)
+	count := 0
+	type args struct {
+		messages []messagebroker.ReceivedMessage
+	}
+	tests := []struct {
+		name    string
+		wantErr bool
+		args    args
+	}{
+		{
+			name: "Test Delay Consumer msg pushed to DLQ success",
+			args: args{
+				messages: msgs,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			consumer.EXPECT().ReceiveMessages(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: tt.args.messages}, nil
+					} else {
+						return &messagebroker.GetMessagesFromTopicResponse{}, nil
+					}
+				}).AnyTimes()
+			go dc.Run(ctx)
+			select {
+			case err := <-errCh:
+				t.Errorf("Got Error : %v", err)
+			case <-time.NewTicker(time.Millisecond * 500).C:
+				cancel()
+				<-time.NewTicker(time.Millisecond * 1).C
+			}
+		})
+	}
+}
+
```

Comment:
there are no assertions here.. how are we checking if a message is pushed to dlq..?